### PR TITLE
Promotions list page

### LINKED ISSUE #109: As an operator, I want to see a status in the listing page that is based on the dates of the promotion and whether it is enable or disabled
The status column in the listing page should show a status badge based the scheduled dates of the promotion, today's date, and whether the promotion is enabled or disabled.

- [ ] User should see Enabled status when a promo is Enabled but has an upcoming Start Date
- [ ] User should see Active status when a promo is Enabled and today's date is within the range of the Start Date and End Date
- [ ] User should see Disabled status when a promo is Disabled and has an End date in the future, or no scheduled dates
- [ ] User should see Past status when a promo has an End Date that is before today

Mockup Link: https://www.figma.com/file/vWot2cXXgjcsLpMq4SAX90/Kinetic?node-id=2020%3A54523

### LINKED ISSUE #111: As an operator I want to be able to see promotions, or a list of only active ones, or a list of “upcoming” promotions
Users should be able to filter the list based on some common presets, including Active, Upcoming, Disabled, View All, and Archived ( if user has See Archived Promotions permission)

- [ ] User can click a header and the list updates to show only promos with the relevant statuses
- [ ] Active view shows promos that are Enabled and have a Start date earlier than today, and an End date later than today
- [ ] Upcoming view shows promos that have a Start date after today
- [ ] Disabled view shows any promos that are currently Disabled
- [ ] View All shows all promos (except Archived)
- [ ] Archived promos should not appear in any view other than Archived

### LINKED ISSUE #113: As an operator I want to see a Promotions Listing page that lists my promotions so that I can easily find the promotion I’m looking for
The purpose of this ticket is to capture the various features of the layout, other tickets will go more in depth on the individual feature functionality (eg Search, Action, Status etc)

**Acceptance Criteria:**

- [ ] User can see a list of Promotions in a table, with columns that display
- Name
- Promo ID
- Promo Type
- Action
- Start Date
- End Date
- Status (issue [6625](https://app.zenhub.com/workspaces/moc-main-board-62fd8e881686130027eafc31/issues/reactioncommerce/reaction/6625))

- [ ] User can access tab navigation that links to filtered views of the table with the following options (issue [6623](https://app.zenhub.com/workspaces/moc-main-board-62fd8e881686130027eafc31/issues/reactioncommerce/reaction/6623))
- Active
- Upcoming
- Disabled
- Past
- All

- [ ] User can access search to find the promo they are looking for

- [ ] User can access filters to find the promo they are looking for

- [ ] User can sort the list ascending / descending by clicking on table headers

- [ ] User can access an Actions dropdown menu (issue [6622](https://app.zenhub.com/workspaces/moc-main-board-62fd8e881686130027eafc31/issues/reactioncommerce/reaction/6622))

- [ ] User can select individual promotions in the table or select All by clicking the checkbox in the header

- [ ] User can navigate the list using standard pagination functionality

- [ ] User can access the Promotion Detail Page by clicking the Promotion Name

Mockup Link: https://www.figma.com/file/vWot2cXXgjcsLpMq4SAX90/Kinetic?node-id=2020%3A54523

### PR DESCRIPTION
**Pull request by vannguyenn**
Resolves #109
Resolves #113
Resolves #111

Testing Instructions:
- Use branch `feat/promotions` on API to test this feature
- Start the app and login with a valid account
- Navigate to Promotions
- Select different tabs to view promotions
Screen Shot 2022-11-17 at 11 16 51
Screen Shot 2022-11-17 at 11 16 42
Screen Shot 2022-11-17 at 11 16 27

- Click Actions button, by default only Create option is enabled. Click on one or multiple row checkboxes to enable these actions. The functionalities of these actions will be implemented in other PR

image

### CODE REVIEW

**Comment by brent-hoover** [comment]
Not directly relevant to this PR but see a couple of typos here on the permission denied screen:

* should be your
* should be either shop manager or owner, "shop owner" is sort of confusing
Email you shop owner to get access

**Review by brent-hoover — APPROVED** [review]
It would be great if we could show the "noop" action, but this is amazing work

**Review by tedraykov — APPROVED** [review]
The PR looks great!

File: src/pages/Promotions/List/index.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  ], []);
+
+  const handleChangeTab = (value: string) => {
+    setSearchParams({ type: value });
+    onRowSelectionChange({});
+  };
+
+  const disabledActionItem = Object.keys(rowSelection).length === 0;
+  return (
+    <Container sx={{ padding: "20px 30px" }} maxWidth={false}>
```
Comment (tedraykov) [inline src/pages/Promotions/List/index.tsx]: I would prefer not to have absolute pixel values. The x-axis padding is already defined by the Container component so no need to set it, and for the y-axis, we can use the mui theme spacing API by setting it to 3 for example.

**Pull request merged**